src/sentry/api/endpoints/project_group_index.py

```diff
@@ -271,6 +274,9 @@ def get(self, request, project):
             self.build_cursor_link(request, 'next', cursor_result.next),
         ])
 
+        if results and query not in [s['query'] for s in DEFAULT_SAVED_SEARCHES]:
```

Comment:
Let's optimize this:

We have two options, either precompute a `set()` of the actual query values, something like:

```python
SAVED_SEARCH_QUERIES = {s['query'] for s in DEFAULT_SAVED_SEARCHES}
```

Then just doing:

```python
query not in SAVED_SEARCH_QUERIES
```

Or, we can short cut the loop here:

```python
for s in DEFAULT_SAVED_SEARCHES:
  if s['query'] == query:
    advanced_search.send(...)
    break
```